Public/Src/Cache/ContentStore/Distributed/Sessions/DistributedContentSession.cs

```diff
@@ -83,10 +89,25 @@ private enum Counters
             UrgencyHint urgencyHint,
             Counter retryCounter)
         {
-            return PutCoreAsync(
-                operationContext,
-                (decoratedStreamSession, wrapStream) => decoratedStreamSession.PutFileAsync(operationContext, path, contentHash, realizationMode, operationContext.Token, urgencyHint, wrapStream),
-                session => session.PutFileAsync(operationContext, contentHash, path, realizationMode, operationContext.Token, urgencyHint));
+            return PerformPutFileGatedOperationAsync(operationContext, () => {
+                return PutCoreAsync(

```

Comment:
>PutCoreAsync [](start = 23, length = 12)

I think it make sense to have a comment here explaining that we intentionally not gating PutStream operations because we don't expect a high number of them at the same time.